### PR TITLE
improved sendmail action

### PR DESCRIPTION
**Pull request by NiekD**
All arguments mandatory
Removed unwanted settings
Added conditionals for EVENTS and WINDOWS
Applied threading
Several small fixes

### CODE REVIEW

File: libs/actions/sendmail.c [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+			nrvalues++;
+			jchild = jchild->next;
+		}
+	}
+	if(nrvalues != 1) {
+		logprintf(LOG_ERR, "sendmail action \"MESSAGE\" only takes one argument");
+		return -1;
+	}
+	nrvalues = 0;
+	if((jto = json_find_member(jmessage, "value")) != NULL) {
```
Comment (CurlyMoo) [inline libs/actions/sendmail.c]: This line is incorrect.

File: libs/actions/sendmail.c [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+				}
+				FREE(mail.from);
+				FREE(mail.to);
+				FREE(mail.subject);
+				FREE(mail.message);
+			}
+		}
+	}
+
+action_sendmail->nrthreads--;
```
Comment (CurlyMoo) [inline libs/actions/sendmail.c]: Missing tab.

File: libs/actions/sendmail.c [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+							
+				mail.from = MALLOC(strlen(sfrom)+1);
+				mail.to = MALLOC(strlen(to)+1);
+				mail.subject = MALLOC(strlen(subject)+1);
+				mail.message = MALLOC(strlen(message)+1);
+			
+				strcpy(mail.from, sfrom);
+				strcpy(mail.to, to);
+				strcpy(mail.subject, subject);
+				strcpy(mail.message, message);
```
Comment (CurlyMoo) [inline libs/actions/sendmail.c]: You can also do (i think):
```
settings_find_string("smtp-sender", &mail.from);
mail.to = jval3->string_;
mail.subject = jval1->string_;
mail.message = jval2->string_;
```
This saves us additional `mallocs` and `frees`.

File: libs/actions/sendmail.c [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+#if !defined(MODULE) && !defined(_WIN32)
+__attribute__((weak))
+#endif
+void actionSendmailInit(void) {
+	event_action_register(&action_sendmail, "sendmail");
+
+	options_add(&action_sendmail->options, 'a', "SUBJECT", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'b', "MESSAGE", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'c', "TO", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
```
Comment (CurlyMoo) [inline libs/actions/sendmail.c]: Can we add the email regex here?

File: libs/actions/sendmail.c [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+	if(jsubject == NULL) {
+		logprintf(LOG_ERR, "sendmail action is missing a \"SUBJECT\"");
+		return -1;
+	}
+	if(jmessage == NULL) {
+		logprintf(LOG_ERR, "sendmail action is missing a \"MESSAGE\"");
+		return -1;
+	}
+	if(jto == NULL) {
+		logprintf(LOG_ERR, "sendmail action is missing a \"TO\"");
```
Comment (CurlyMoo) [inline libs/actions/sendmail.c]: Please validate the mail address somewhere.

File: libs/actions/sendmail.c (hunk 1 above)
Comment (NiekD) [inline libs/actions/sendmail.c]: Oops...

CurlyMoo schreef op 21-3-2015 om 15:29:

File: libs/actions/sendmail.c (hunk 3 above)
Comment (NiekD) [inline libs/actions/sendmail.c]: I agree.

CurlyMoo schreef op 21-3-2015 om 15:33:

File: libs/actions/sendmail.c (hunk 4 above)
Comment (NiekD) [inline libs/actions/sendmail.c]: Yes.

CurlyMoo schreef op 21-3-2015 om 15:34:

File: libs/actions/sendmail.c (hunk 5 above)
Comment (NiekD) [inline libs/actions/sendmail.c]: OK.

CurlyMoo schreef op 21-3-2015 om 15:34:

**Pull request closed without merging**